fda/fdata_boxplot.py

```diff
@@ -0,0 +1,627 @@
+"""Functional Data Boxplot Module.
+
+This module contains the classes to construct the functional data boxplot and
+visualize it.
+
+"""
+import matplotlib
+import matplotlib.pyplot as plt
+import math
+
+from .depth_measures import *
+from .grid import FDataGrid
+from io import StringIO
+from abc import ABC, abstractmethod
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+class FDataBoxplot(ABC):
+    """Abstract class inherited by the Boxplot and SurfaceBoxplot classes.
+
+    It the data of the functional boxplot or surface boxplot of a FDataGrid object,
+    depending on the dimensions of the domain, 1 or 2 respectively.
+
+    It forces to both classes, Boxplot and SurfaceBoxplot to conain at least the median,
+    central and outlying envelopes and a colormap for their graphical representation,
+    obtained calling the plot method.
+
+    """
+    @abstractmethod
+    def __init__(self, factor=1.5):
+        if factor < 0:
+            raise ValueError(
+                "The number used to calculate the outlying envelope must be positive.")
+
+    @property
+    def fdatagrid(self):
+        pass
+
+    @property
+    def median(self):
+        pass
+
+    @property
+    def central_envelope(self):
+        pass
+
+    @property
+    def outlying_envelope(self):
+        pass
+
+    @property
+    def colormap(self):
```

Comment:
If the colormap is the same for all Boxplot types, you can implement it in the abstract class instead.